app/src/main/java/io/gnosis/safe/ui/transaction/list/TransactionPagingSource.kt

```diff
@@ -0,0 +1,40 @@
+package io.gnosis.safe.ui.transaction.list
+
+import androidx.paging.PagingSource
+import io.gnosis.data.models.SafeInfo
+import io.gnosis.data.models.Transaction
+import io.gnosis.data.repositories.SafeRepository
+import io.gnosis.data.repositories.TransactionRepository
+import pm.gnosis.model.Solidity
+import retrofit2.HttpException
+import java.io.IOException
+
+
+class TransactionPagingSource(
+    private val safe: Solidity.Address,
+    private val safeRepo: SafeRepository,
+    private val txRepo: TransactionRepository
+) : PagingSource<String, Transaction>() {
+
+    private lateinit var safeInfo: SafeInfo
+
+    override suspend fun load(params: LoadParams<String>): LoadResult<String, Transaction> {
+
+        val pageLink = params.key
+
+        safeInfo = safeRepo.getSafeInfo(safe)
+
+        return try {
+            val response = pageLink?.let { txRepo.loadTransactionsPage(pageLink, safeInfo) } ?: txRepo.getTransactions(safe, safeInfo, params.loadSize)
+            LoadResult.Page(
+                data = response.results,
+                prevKey = response.previous,
+                nextKey = response.next
+            )
+        } catch (exception: IOException) {
+            return LoadResult.Error(exception)
+        } catch (exception: HttpException) {
+            return LoadResult.Error(exception)
+        }
```

Comment:
Could this be written as a `runCatching` block? It's not a big deal anyway xD it just look a bit prettier :)